File: multi/infrastructure/src/main/java/com/example/infrastructure/movie/MovieRepositoryImpl.java
package com.example.infrastructure.movie;

import com.example.domain.movies.entity.Movie;
import com.example.domain.movies.service.MovieRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
@Slf4j
public class MovieRepositoryImpl implements MovieRepository {

    private final MovieJpaRepository movieJpaRepository;

    @Override
    public Movie getMovieInfo(Long movieId) {
        return movieJpaRepository.findById(movieId).orElse(null);
    }
}

File: movie/infra/src/main/kotlin/com/example/movie/repository/MovieRepositoryImpl.kt
package com.example.movie.repository

import com.example.movie.domain.movie.model.Movie
import com.example.movie.domain.movie.repository.MovieRepository
import org.springframework.stereotype.Repository

@Repository
class MovieRepositoryImpl(
    private val movieJpaRepository: MovieJpaRepository
) : MovieRepository {
    override fun findById(id: Long): Movie? {
        return movieJpaRepository.findById(id).orElse(null)?.toDomain()
    }

    override fun findAllOrderByReleaseDateDesc(): List<Movie> {
        return movieJpaRepository.findAllByOrderByReleaseDateDesc().map { it.toDomain() }
    }
}